kornia/losses/perceptual.py

```diff
@@ -0,0 +1,324 @@
+from typing import Callable, Tuple
+
+import torch
+import torch.nn as nn
+import torchvision
+
+import kornia
+from kornia import testing
+from kornia.geometry.transform.imgwarp import get_perspective_transform, warp_perspective
+
+
+class LossNetwork(nn.Module):
+
+    def __init__(
+        self,
+        model_name: str = 'resnet34',
+        output_layer: int = 1,
+        freeze: bool = True,
+    ) -> None:
+        super().__init__()
+        if 'resnet' not in model_name:
+            raise ValueError('Only ResNets are supported ATM')
+
+        # Define resnet model
+        loss_network_fn = getattr(torchvision.models, model_name)
+        self.resnet = loss_network_fn(pretrained=True, progress=False)
```

Comment:
instead of this, which add `torchvision` as dep, why not pass the resnet directly as a parameter?